src/AutoRest.CSharp/Mgmt/Generation/ManagementGroupExtensionsWriter.cs

```diff
@@ -45,13 +46,13 @@ public override void WriteExtension(CodeWriter writer, BuildContext<MgmtOutputLi
             }
         }
 
-        private void WriteGetResourceContainerMethod(CodeWriter writer, ResourceContainer resourceContainer)
+        private void WriteGetResourceContainerMethod(CodeWriter writer, Resource armResource, ResourceContainer resourceContainer)
         {
             writer.WriteXmlDocumentationSummary($"Gets an object representing a {resourceContainer.Type.Name} along with the instance operations that can be performed on it.");
             writer.WriteXmlDocumentationParameter(ExtensionOperationVariableName, $"The <see cref=\"{typeof(ManagementGroupOperations)}\" /> instance the method will execute against.");
             writer.WriteXmlDocumentationReturns($"Returns a <see cref=\"{resourceContainer.Type.Name}\" /> object.");
 
-            using (writer.Scope($"public static {resourceContainer.Type} Get{resourceContainer.Type.Name}(this {typeof(ManagementGroupOperations)} {ExtensionOperationVariableName})"))
+            using (writer.Scope($"public static {resourceContainer.Type} Get{armResource.Type.Name.ToPlural()}(this {typeof(ManagementGroupOperations)} {ExtensionOperationVariableName})"))
```

Comment:
Instead of passing the `resource` in, actually you can just use `resourceContainer.Resource` to get its underlying `Resource` type.

Reply:
Updated.